### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by Darruma**

### CODE REVIEW

File: integration_tests/test_cycles_eth.py [hunk 1]
```diff
@@ -26,23 +26,27 @@
...
+    )
 
 
 @pytest.fixture(autouse=True)
-def set_env_config(monkeypatch):
+def set_env_config(mocker):
     env_config.test = False
     env_config.staging = False
     env_config.production = True
+    env_config.get_web3 = lambda chain: web3
```
Comment (SHAKOTN) [inline integration_tests/test_cycles_eth.py:49]: What's the purpose of `lambda chain: web3`?

File: rewards/classes/TreeManager.py [hunk 2]
```diff
@@ -204,19 +201,7 @@ def last_propose_start_block(self) -> int:
         return self.badger_tree.lastProposeStartBlock().call()
 
     def get_tx_options(self, account: Account) -> dict:
```
Comment (SHAKOTN) [inline rewards/classes/TreeManager.py:203]: This is redundant I guess? Why just not call `create_tx_options(account.address, self.w3, self.chain)` directly ?

File: rewards/utils/tx_utils.py [hunk 3]
```diff
@@ -194,3 +194,26 @@ def confirm_transaction(
...
+
+def create_tx_options(address, w3, chain):
+    options = {
+        "nonce": w3.eth.get_transaction_count(address),
+        "from": address,
+    }
+    if chain == Network.Ethereum:
+        options["maxPriorityFeePerGas"] = get_priority_fee(w3)
+        options["maxFeePerGas"] = get_effective_gas_price(w3, chain)
+        options["gas"] = 200000
```
Comment (SHAKOTN) [inline rewards/utils/tx_utils.py:207]: Should be placed in constants

File: rewards/utils/tx_utils.py [hunk 4]
```diff
@@ -194,3 +194,26 @@ def confirm_transaction(
...
+    elif chain == Network.Arbitrum:
+        options["gas"] = 3000000
```
Comment (SHAKOTN) [inline rewards/utils/tx_utils.py:209]: Should be placed in constants

File: rewards/utils/tx_utils.py [hunk 5]
```diff
@@ -194,3 +194,26 @@ def confirm_transaction(
         msg = f"Error waiting for {tx_hash}. Error: {e}."
         logger.error(msg)
         return False, msg
+
+
+def create_tx_options(address, w3, chain):
```
Comment (SHAKOTN) [inline rewards/utils/tx_utils.py:199]: Some typing would be useful

File: rewards/utils/tx_utils.py [hunk 6]
```diff
@@ -194,3 +194,26 @@ def confirm_transaction(
...
+        options["maxFeePerGas"] = get_effective_gas_price(w3, chain)
+        options["gas"] = 200000
+    elif chain == Network.Arbitrum:
+        options["gas"] = 3000000
+    else:
+        options["gasPrice"] = get_effective_gas_price(w3, chain)
+    return options
+
+
+def build_and_send(func, options, w3, pkey):
```
Comment (SHAKOTN) [inline rewards/utils/tx_utils.py:215]: Some typing please ser

File: tests/test_managers/test_tree_manager.py [hunk 7]
```diff
@@ -55,6 +59,17 @@ def test_matches_pending_hash(tree_manager):
     assert not tree_manager.matches_pending_hash(random_hash)
 
 
+@pytest.mark.parametrize(
+    "tree_manager",
+    [Network.Ethereum],
+    indirect=True,
+)
+def test_build_function_and_send(tree_manager, mocker):
```
Comment (SHAKOTN) [inline tests/test_managers/test_tree_manager.py:67]: What exactly does this test? Maybe check that some mocked functions were called?

File: tests/test_utils/test_tx_utils.py [hunk 8]
```diff
@@ -118,6 +121,28 @@ def test_get_latest_base_fee(fee):
     assert get_latest_base_fee(web3) == int(fee)
 
 
+@pytest.mark.parametrize("chain", [Network.Ethereum, Network.Arbitrum, Network.Fantom])
+def test_create_tx_options(mocker, chain):
```
Comment (SHAKOTN) [inline tests/test_utils/test_tx_utils.py:125]: This is some good test ser

File: integration_tests/test_cycles_eth.py (hunk 1 above)
Comment (SHAKOTN) [reply integration_tests/test_cycles_eth.py:49]: Yes, but why you need to pass `chain` as param to lambda?

Comment (Darruma) [reply integration_tests/test_cycles_eth.py:49]: because the get_web3 function has that as the param

**Review by SHAKOTN — APPROVED** [review]

**Pull request merged**